.circleci/config.yml

```diff
@@ -0,0 +1,65 @@
+version: 2.1
+
+references:
+  workdir: &workdir
+    working_directory: ~/source
+
+  persist-workspace: &persist-workspace
+    persist_to_workspace:
+      root: ~/source
+      paths:
+        - .
+
+  attach-workspace: &attach-workspace
+    attach_workspace:
+      at: .
+
+jobs:
+  build:
+    <<: *workdir
+
+    docker: # run the steps with Docker
+      - image: circleci/openjdk:8-jdk-stretch
+
+    steps:
+      - checkout
+      - restore_cache:
+          keys:
+            - v2-dependencies-{{ checksum "pom.xml" }}
+            # fallback to using the latest cache if no exact match is found
+            - v2-dependencies-
+      - run: mvn -s .circleci/.circleci.settings.xml clean install
+      # Built module is stored in: /home/circleci/source/repo/repo/api/target/jcontent-2.0.0-SNAPSHOT.jar
+      - save_cache:
+          paths:
+            - ~/.m2
+          key: v2-dependencies-{{ checksum "pom.xml" }}
+      - run:
+          name: Prepare artifacts
+          command: |
+            mkdir /tmp/artifacts/
+            cp /home/circleci/source/target/*.jar /tmp/artifacts/
+      - store_artifacts:
+          path: /tmp/artifacts/
+      - *persist-workspace
+  deploy:
+    <<: *workdir
+
+    docker: # run the steps with Docker
+      - image: circleci/openjdk:8-jdk-stretch
+    
+    steps:
+      - *attach-workspace
+      - run: mvn -s .circleci/.circleci.settings.xml deploy
+
+workflows:
+  version: 2
+  build:
+    jobs:
+      - build
```

Comment:
You should use context here, this way we can store nexus credentials in one single place.

I just realized this was dropped (probably a mistake on my end) from jahia-ui-root. 
You can specify the context like this:
``` 
- build:
    context: QA_ENVIRONMENT
```   
Documentation is available here: https://circleci.com/docs/2.0/contexts/

I'll add it back to jahia-ui-root, because as-is it's only using the maven cache.

Reply:
So I can then remove the settings file ?

Reply:
you still need the settings file, but the environment variables would be coming from the context instead of the local project env vars

Reply:
OK so I just have to remove the variable from the project settings

Reply:
yes, I'd also clear the maven cache just to be sure settings are properly taken in.